earn/src/components/info/InfoGraph.tsx

```diff
@@ -0,0 +1,177 @@
+
+      flattenedGraphData.push(entry);
+    })
+  );
+
+  flattenedGraphData.sort((a, b) => a.x.getTime() - b.x.getTime());
+
+  const displayedGraphData: { [k: InfoGraphLabel]: number; x: number }[] = [];
+  for (let i = 0; i < flattenedGraphData.length; i++) {
+    const entry = { ...flattenedGraphData[i], x: flattenedGraphData[i].x.getTime() };
+
+    if (entry.x !== displayedGraphData.at(-1)?.x) {
+      displayedGraphData.push({
+        ...(displayedGraphData.at(-1) ?? {}),
+        ...entry,
+      });
+      continue;
+    }
+
+    const previousEntry = displayedGraphData.at(-1)!;
+    Object.assign(previousEntry, entry);
+  }
+
+  const charts: GraphChart[] = useMemo(
+    () =>
+      labels.map((label) => {
+        const shouldBeColored =
+          hoveredPair === undefined || label === `${hoveredPair!.token0.symbol}/${hoveredPair!.token1.symbol}`;
+        let stroke = 'rgba(43, 64, 80, 0.5)';
+        if (shouldBeColored) {
+          stroke = graphColors.has(label) ? `url(#${label.replace('/', '-')})` : 'white';
+        }
+
+        const shouldBeThick = shouldBeColored && hoveredPair !== undefined;
+
+        return {
+          uniqueId: label,
+          dataKey: label,
+          stroke,
+          strokeWidth: shouldBeThick ? 4 : 2,
+          type: 'monotone',
+        };
+      }),
+    [labels, graphColors, hoveredPair]
+  );
+
+  const linearGradients = useMemo(() => {
+    const arr: SVGProps<SVGLinearGradientElement>[] = [];
+
+    graphColors.forEach((v, k) => {
+      arr.push(
+        <linearGradient id={k.replace('/', '-')} x1='0' y1='0' x2='0' y2='1'>
+          <stop offset='-29%' stopColor={v.color0} stopOpacity={1} />
+          <stop offset='75%' stopColor={v.color1} stopOpacity={1} />
+        </linearGradient>
+      );
+    });
+
+    return arr;
+  }, [graphColors]);
+
+  return (
+    <TableContainer>
+      <Table>
+        <thead className='text-start'>
+          <tr>
+            <TableHeaderElement className='px-4 py-2 text-center whitespace-nowrap'>
+              <Text size='M' weight='bold'>
+                LTV History
+              </Text>
+            </TableHeaderElement>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <td>
+              {graphData && (
+                <LineGraph
+                  linearGradients={linearGradients}
+                  CustomTooltip={<InfoGraphTooltip />}
+                  tooltipPosition={{ x: 0, y: 0 }}
+                  charts={charts}
+                  data={displayedGraphData}
+                  size={
+                    isTabletOrBigger
+                      ? {
+                          width: Number(FULL_WIDTH),
+                          height: Number(FULL_HEIGHT) - 48,
+                        }
+                      : {
+                          width: windowDimensions.width - 32,
+                          height: Number(MOBILE_HEIGHT) - 48,
+                        }
+                  }
```

Comment:
There's probably a better way to handle this, but recharts sizing (even with `ResponsiveContainer`) is super finicky.